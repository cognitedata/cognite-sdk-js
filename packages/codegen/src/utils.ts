// Copyright 2022 Cognite AS
import path from 'path';
import { promises as fs } from 'fs';

export type FromPathOption = {
  'from-path': string;
};

export type PackageOption = {
  package: string;
};

export type ServiceOption = {
  service: string;
};

export type VersionOption = {
  version: string;
};

const createPath = (
  options: PackageOption & Partial<ServiceOption>
): string => {
  const packagePath = path.resolve('./packages', options.package, 'src');

  if (typeof options.service !== 'undefined') {
    return path.resolve(packagePath, 'api', options.service);
  }
  return packagePath;
};

export const versionFileDirectoryPath = async (
  options: PackageOption & Partial<ServiceOption>
): Promise<string> => {
  const directory = createPath(options);
  try {
    await fs.stat(directory);
    return directory;
  } catch (error) {
    throw new Error(`Path "${directory}" does not exist: ${error}`);
  }
};
