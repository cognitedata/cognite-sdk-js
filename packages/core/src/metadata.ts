// Copyright 2020 Cognite AS
import type { HttpResponse } from './httpClient/basicHttpClient';

export interface Metadata {
  status: number;
  headers: { [key: string]: string };
}

/** @hidden */
export class MetadataMap {
  private map: WeakMap<object, Metadata>;
  constructor() {
    this.map = new WeakMap();
  }

  public addAndReturn<T extends object, V>(
    value: T,
    metadata: HttpResponse<V>,
  ): T {
    this.map.set(value, {
      // we extract out only what is necessary
      status: metadata.status,
      headers: metadata.headers,
    });
    return value;
  }

  public get(value: object): undefined | Metadata {
    return this.map.get(value);
  }
}
