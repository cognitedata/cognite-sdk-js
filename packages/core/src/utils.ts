// Copyright 2020 Cognite AS

import isObject from 'lodash/isObject';
import isArray from 'lodash/isArray';
import isBuffer from 'is-buffer';
import { API_VERSION, BASE_URL } from './constants';
import { CogniteError } from './error';
import { CogniteMultiError } from './multiError';

/** @hidden */
export function getBaseUrl(baseUrl?: string) {
  return baseUrl || BASE_URL;
}

/** @hidden */
export const apiUrl = () => `/api/${API_VERSION}`;

/** @hidden */
export function projectUrl(project: string) {
  return `${apiUrl()}/projects/${encodeURIComponent(project)}`;
}

/** @hidden */
export function bearerString(token: string) {
  return `Bearer ${token}`;
}

/** @hidden */
export function isBrowser() {
  return (
    typeof window !== 'undefined' && typeof window.document !== 'undefined'
  );
}

export function clearParametersFromUrl(...params: string[]): void {
  let url = window.location.href;
  params.forEach(param => {
    url = removeQueryParameterFromUrl(url, param);
  });
  window.history.replaceState(null, '', url);
}

/** @hidden */
export function removeQueryParameterFromUrl(
  url: string,
  parameter: string
): string {
  return url
    .replace(new RegExp('[?#&]' + parameter + '=[^&#]*(#.*)?$'), '$1')
    .replace(new RegExp('([?#&])' + parameter + '=[^&]*&'), '$1');
}

/** @hidden */
export function convertToTimestampToDateTime(timestamp: number): Date {
  return new Date(timestamp);
}

/** @hidden */
export function isJson(data: any) {
  return (
    (isArray(data) || isObject(data)) &&
    !isBuffer(data) &&
    !(data instanceof ArrayBuffer) &&
    !(data instanceof Date)
  );
}

/** @hidden */
export function sleepPromise(durationInMs: number) {
  return new Promise(resolve => {
    setTimeout(resolve, durationInMs);
  });
}

/** @hidden */
export function promiseCache<ReturnValue>(
  promiseFn: () => Promise<ReturnValue>
) {
  let unresolvedPromise: Promise<ReturnValue> | null = null;
  return () => {
    if (unresolvedPromise) {
      return unresolvedPromise;
    }
    return (unresolvedPromise = promiseFn().then(res => {
      unresolvedPromise = null;
      return res;
    }));
  };
}

/**
 * Resolves all Promises at once, even if some of them fail
 */
/** @hidden */
export async function promiseAllAtOnce<RequestType, ResponseType>(
  inputs: RequestType[],
  promiser: (input: RequestType) => Promise<ResponseType>
) {
  const failed: RequestType[] = [];
  const succeded: RequestType[] = [];
  const responses: ResponseType[] = [];
  const errors: (Error | CogniteError)[] = [];

  const promises = inputs.map(input => promiser(input));

  const wrappedPromises = promises.map((promise, index) =>
    promise
      .then(result => {
        succeded.push(inputs[index]);
        responses.push(result);
      })
      .catch(error => {
        failed.push(inputs[index]);
        errors.push(error);
      })
  );
  await Promise.all(wrappedPromises);
  if (failed.length) {
    throw {
      succeded,
      responses,
      failed,
      errors,
    };
  }

  return responses;
}

/** @hidden */
export async function promiseAllWithData<RequestType, ResponseType>(
  inputs: RequestType[],
  promiser: (input: RequestType) => Promise<ResponseType>,
  runSequentially: boolean
) {
  try {
    if (runSequentially) {
      return await promiseEachInSequence(inputs, promiser);
    } else {
      return await promiseAllAtOnce(inputs, promiser);
    }
  } catch (err) {
    throw new CogniteMultiError(err);
  }
}

/**
 * Resolves Promises sequentially
 */
/** @hidden */
export async function promiseEachInSequence<RequestType, ResponseType>(
  inputs: RequestType[],
  promiser: (input: RequestType) => Promise<ResponseType>
) {
  return inputs.reduce(async (previousPromise, input, index) => {
    const prevResult = await previousPromise;
    try {
      return prevResult.concat(await promiser(input));
    } catch (err) {
      throw {
        errors: [err],
        failed: inputs.slice(index),
        succeded: inputs.slice(0, index),
        responses: prevResult,
      };
    }
  }, Promise.resolve(new Array<ResponseType>()));
}

/** @hidden */
export function isSameProject(project1: string, project2: string): boolean {
  return project1.toLowerCase() === project2.toLowerCase();
}

/** @hidden */
export function applyIfApplicable<ArgumentType, ResultType>(
  args: ArgumentType,
  action?: (input: ArgumentType) => ResultType
) {
  if (action) {
    return action(args);
  }
  return args;
}

/** @hidden */
export function generatePopupWindow(url: string, name: string) {
  return window.open(
    url,
    name,
    // https://www.quackit.com/javascript/popup_windows.cfm
    'height=500,width=400,left=100,top=100,resizable=yes,scrollbars=yes,toolbar=yes,menubar=no,location=no,directories=no, status=yes'
  );
}

/** @hidden */
export function createInvisibleIframe(
  url: string,
  name: string
): HTMLIFrameElement {
  const iframe = document.createElement('iframe');
  iframe.name = name;
  iframe.style.width = '0';
  iframe.style.height = '0';
  iframe.style.border = '0';
  iframe.style.border = 'none';
  iframe.style.visibility = 'hidden';

  iframe.setAttribute('id', name);
  iframe.setAttribute('aria-hidden', 'true');

  iframe.src = url;
  return iframe;
}

/** @hidden */
export function isUsingSSL() {
  return isBrowser() && location.protocol.toLowerCase() === 'https:';
}

/** @hidden */
export function isLocalhost(): boolean {
  return isBrowser() && location.hostname === 'localhost';
}
