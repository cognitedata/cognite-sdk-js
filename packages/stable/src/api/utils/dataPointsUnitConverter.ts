import {
  Datapoints,
  StringDatapoint,
  DoubleDatapoint,
  UnitConverterOptions,
} from '../../types';

import convertUnit from './unitConverter';

/**
 * Convert datapoints based on the given output unit.
 * Note that there are two possible behaviors if the conversion fails:
 * It will depend on `unitsConversionOptions.continueIfConversionFails` parameter. If it's true
 * the application will throw an error and stop running if conversion fails. If it's false it will log warns.
 * A fail might occur if there is something wrong with the datapoint value or the selected output unit
 * is unsuported.
 * @param dataPointsData
 * @param unitsConversionOptions
 * @returns dataPointsData that were possible to convert
 */
const dataPointsUnitsConverter = (
  dataPointsData: Datapoints[],
  unitsConversionOptions: UnitConverterOptions
): Datapoints[] => {
  const converted = dataPointsData?.map((data) =>
    processDataPointsData(data, unitsConversionOptions)
  );

  return converted;
};

function processDataPointsData(
  data: Datapoints,
  unitsConversionOptions: UnitConverterOptions
): Datapoints {
  if (data.unit && unitsConversionOptions.continueIfConversionFails) {
    try {
      return convertDataPoints(data, unitsConversionOptions);
    } catch (e) {
      console.warn(
        `
          Something went wrong when converting from ${data.unit}
          to ${unitsConversionOptions.outputUnit}
          for datapoints with ${
            data.externalId ? `externalId ${data.externalId}` : `id ${data.id}`
          }
        `
      );
    }
  } else if (data.unit && !unitsConversionOptions.continueIfConversionFails) {
    return convertDataPoints(data, unitsConversionOptions);
  }

  return data;
}

/**
 * Tries to convert datapoints units. If any fails it gives up converting and returns
 * the @data input param
 * @param data
 * @param unitsConversionOptions
 * @returns converted data points
 */
function convertDataPoints(
  data: Datapoints,
  unitsConversionOptions: UnitConverterOptions
): Datapoints {
  const convertedDataPoints: StringDatapoint[] | DoubleDatapoint[] =
    processDataPoints(data, unitsConversionOptions);

  const newData = Object.assign({}, data);
  newData.datapoints = convertedDataPoints;

  return newData;
}

function processDataPoints(
  data: Datapoints,
  unitsConversionOptions: UnitConverterOptions
): StringDatapoint[] | DoubleDatapoint[] {
  const datapoints: (StringDatapoint | DoubleDatapoint)[] = [];

  for (const dataPoint of data.datapoints) {
    const convertedValue = convertUnit(
      dataPoint.value,
      data.unit!,
      unitsConversionOptions.outputUnit,
      unitsConversionOptions.precision
    );

    const newDataPoint = Object.assign({}, dataPoint);
    newDataPoint.value = convertedValue;

    datapoints.push(newDataPoint);
  }

  return datapoints as StringDatapoint[] | DoubleDatapoint[];
}

export default dataPointsUnitsConverter;
