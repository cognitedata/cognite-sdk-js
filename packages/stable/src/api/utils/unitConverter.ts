import {
  Converter,
  UnitDictionariesProcessorImpl,
} from '@cognite-ornellas/units/';

const converter = new Converter(new UnitDictionariesProcessorImpl());

const unitConverter = (
  value: any,
  inputUnit: string,
  outputUnit: string,
  precision = 6
) => {
  return converter
    .unitConvert(value, inputUnit, outputUnit)
    .toPrecision(precision);
};

export default unitConverter;
