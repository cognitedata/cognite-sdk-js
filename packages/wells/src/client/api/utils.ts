import { CogniteGeospatialClient } from '@cognite/geospatial-sdk-js';
import { Constants } from '@cognite/sdk-core';
//import CogniteClient from '../cogniteClient';

export const geospatialClient = CogniteGeospatialClient({
  project: process.env.COGNITE_WELLS_PROJECT as string,
  api_key: process.env.COGNITE_WELLS_CREDENTIALS as string,
  api_url: Constants.BASE_URL,
});
