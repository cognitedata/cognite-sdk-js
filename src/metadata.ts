// Copyright 2019 Cognite AS
import { HttpResponse } from './utils/http/basicHttpClient';

export interface Metadata {
  status: number;
  headers: { [key: string]: string };
}

/** @hidden */
export class MetadataMap {
  private map: WeakMap<any, Metadata>;
  constructor() {
    this.map = new WeakMap();
  }

  public addAndReturn<T, V>(value: T, metadata: HttpResponse<V>): T {
    this.map.set(value, {
      // we extract out only what is necessary
      status: metadata.status,
      headers: metadata.headers,
    });
    return value;
  }

  public get(value: any): undefined | Metadata {
    return this.map.get(value);
  }
}
