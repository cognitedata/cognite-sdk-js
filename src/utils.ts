// Copyright 2019 Cognite AS

import { cloneDeepWith, isArray, isObject } from 'lodash';
import { isBuffer } from 'util';
import { API_VERSION, BASE_URL } from './constants';

/** @hidden */
export function getBaseUrl(baseUrl?: string) {
  return baseUrl || BASE_URL;
}

/** @hidden */
export const apiUrl = () => `/api/${API_VERSION}`;

/** @hidden */
export function projectUrl(project: string) {
  return `${apiUrl()}/projects/${encodeURIComponent(project)}`;
}

/** @hidden */
export function bearerString(token: string) {
  return `Bearer ${token}`;
}

/** @hidden */
export function isBrowser() {
  return (
    typeof window !== 'undefined' && typeof window.document !== 'undefined'
  );
}

/** @hidden */
export function removeQueryParameterFromUrl(
  url: string,
  parameter: string
): string {
  return url
    .replace(new RegExp('[?&]' + parameter + '=[^&#]*(#.*)?$'), '$1')
    .replace(new RegExp('([?&])' + parameter + '=[^&]*&'), '$1');
}

/** @hidden */
export function convertToTimestampToDateTime(timestamp: number): Date {
  return new Date(timestamp);
}

/** @hidden */
export function isJson(data: any) {
  return (isArray(data) || isObject(data)) && !isBuffer(data);
}

/** @hidden */
export function sleepPromise(durationInMs: number) {
  return new Promise(resolve => {
    setTimeout(resolve, durationInMs);
  });
}

/** @hidden */
export function promiseCache<ReturnValue>(
  promiseFn: () => Promise<ReturnValue>
) {
  let unresolvedPromise: Promise<ReturnValue> | null = null;
  return () => {
    if (unresolvedPromise) {
      return unresolvedPromise;
    }
    return (unresolvedPromise = promiseFn().then(res => {
      unresolvedPromise = null;
      return res;
    }));
  };
}

/** @hidden */
export function isSameProject(project1: string, project2: string): boolean {
  return project1.toLowerCase() === project2.toLowerCase();
}

/** @hidden */
export function applyIfApplicable<ArgumentType, ResultType>(
  args: ArgumentType,
  action?: (input: ArgumentType) => ResultType
) {
  if (action) {
    return action(args);
  }
  return args;
}

/** @hidden */
export function transformDateInRequest<T>(data: T): T {
  return cloneDeepWith(data, value => {
    if (value instanceof Date) {
      return value.getTime();
    }
  });
}

/** @hidden */
export function transformDateInResponse<T>(data: T): T {
  const dateKeys = [
    'createdTime',
    'lastUpdatedTime',
    'uploadedTime',
    'deletedTime',
    'timestamp',
  ];
  return cloneDeepWith(data, (value, key) => {
    if (dateKeys.indexOf('' + key) !== -1) {
      return new Date(value);
    }
  });
}

/** @hidden */
export function generatePopupWindow(url: string, name: string) {
  return window.open(
    url,
    name,
    // https://www.quackit.com/javascript/popup_windows.cfm
    'height=500,width=400,left=100,top=100,resizable=yes,scrollbars=yes,toolbar=yes,menubar=no,location=no,directories=no, status=yes'
  );
}

/** @hidden */
export function createInvisibleIframe(
  url: string,
  name: string
): HTMLIFrameElement {
  const iframe = document.createElement('iframe');
  iframe.name = name;
  iframe.style.width = '0';
  iframe.style.height = '0';
  iframe.style.border = '0';
  iframe.style.border = 'none';
  iframe.src = url;
  return iframe;
}
