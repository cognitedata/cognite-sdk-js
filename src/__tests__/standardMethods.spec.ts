// Copyright 2019 Cognite AS

import axios, { AxiosRequestConfig } from 'axios';
import MockAdapter from 'axios-mock-adapter';
import { MetadataMap } from '../metadata';
import {
  generateCreateEndpoint,
  generateDeleteEndpoint,
  generateInsertEndpoint,
  generateListEndpoint,
  generateRetrieveEndpoint,
  generateUpdateEndpoint,
} from '../standardMethods';
import { sleepPromise } from '../utils';

function numberGeneratorAPIEndpoint(config: AxiosRequestConfig) {
  const params = config.params || {};
  const cursor = Number(params.cursor);
  if (cursor > 10) {
    return [200, { items: [cursor] }];
  } else if (isNaN(cursor)) {
    return [200, { items: [0, 1], nextCursor: '2' }];
  } else {
    return [
      200,
      {
        items: [cursor, cursor + 1],
        nextCursor: `${cursor + 2}`,
      },
    ];
  }
}

function countChunks(axiosMock: MockAdapter, url: string) {
  const counterObject = { value: 0 };
  axiosMock.onPost(url).reply(config => {
    counterObject.value++;
    const { items: requestItems } = JSON.parse(config.data);
    const responseItems = requestItems.map((item: any) => ({
      ...item,
      id: Number(item.name),
    }));
    return [201, { items: responseItems }];
  });
  return counterObject;
}

function fillArray(): { id: number }[] {
  return new Array(2003).fill(null).map(index => ({
    id: index,
  }));
}

describe('standard methods', () => {
  const axiosInstance = axios.create();
  const axiosMock = new MockAdapter(axiosInstance);
  beforeEach(() => {
    axiosMock.reset();
  });

  describe('generateCreateEndpoint', () => {
    test('automatic chunking for create', async () => {
      const metadataMap = new MetadataMap();
      const object = await chunkingTester(
        axiosMock,
        metadataMap,
        'path',
        generateCreateEndpoint(axiosInstance, 'path', metadataMap)
      );
      expect(object.response.length).toBe(object.items.length);
    });
  });
  describe('generateRetrieveEndpoint', () => {
    test('automatic chunking for retrieve', async () => {
      const metadataMap = new MetadataMap();
      const object = await chunkingTester(
        axiosMock,
        metadataMap,
        'path/byids',
        generateRetrieveEndpoint(axiosInstance, 'path', metadataMap)
      );
      expect(object.response.length).toBe(object.items.length);
    });
  });

  describe('generateDeleteEndpoint', () => {
    test('automatic chunking for delete', async () => {
      const metadataMap = new MetadataMap();
      const { response } = await chunkingTester(
        axiosMock,
        metadataMap,
        'path/delete',
        generateDeleteEndpoint(axiosInstance, '/path', metadataMap)
      );
      expect(response).toMatchObject({});
    });
  });
  describe('generateUpdateEndpoint', () => {
    test('automatic chunking for update', async () => {
      const metadataMap = new MetadataMap();
      const object = await chunkingTester(
        axiosMock,
        metadataMap,
        'path/update',
        generateUpdateEndpoint(axiosInstance, 'path', metadataMap)
      );
      expect(object.response.length).toBe(object.items.length);
    });
  });
  describe('generateInsertEndpoint', () => {
    test('automatic chunking for insert', async () => {
      const metadataMap = new MetadataMap();
      const { response } = await chunkingTester(
        axiosMock,
        metadataMap,
        'path',
        generateInsertEndpoint(axiosInstance, '/path', metadataMap)
      );
      expect(response).toMatchObject({});
    });
  });
  describe('generateListEndpoint', () => {
    function createMockedListEndpoint() {
      const listEndpoint = generateListEndpoint<any, number>(
        axiosInstance,
        '/',
        new MetadataMap(),
        false
      );
      axiosMock.onGet('/').reply(numberGeneratorAPIEndpoint);
      return listEndpoint;
    }
    test('for-await', async () => {
      const listEndpoint = createMockedListEndpoint();
      const numbers = [];
      for await (const item of listEndpoint()) {
        numbers.push(item);
      }
      expect(numbers).toMatchSnapshot();
    });

    test('autoPagingToArray', async () => {
      const listEndpoint = createMockedListEndpoint();
      await expect(
        listEndpoint().autoPagingToArray({ limit: 1 })
      ).resolves.toMatchSnapshot();
      await expect(
        listEndpoint().autoPagingToArray({ limit: 3 })
      ).resolves.toMatchSnapshot();
      await expect(
        listEndpoint().autoPagingToArray({ limit: 100 })
      ).resolves.toMatchSnapshot();
    });

    test('authPagingForEach', async () => {
      const listEndpoint = createMockedListEndpoint();

      // sync handler
      {
        const numbers: number[] = [];
        await listEndpoint().autoPagingEach(item => {
          numbers.push(item);
          if (numbers.length >= 3) {
            return false;
          }
        });
        expect(numbers).toMatchSnapshot();
      }

      // async handler
      {
        const numbers: number[] = [];
        await listEndpoint().autoPagingEach(async item => {
          numbers.push(item);
          await sleepPromise(10);
          if (numbers.length >= 3) {
            return false;
          }
        });
        expect(numbers).toMatchSnapshot();
      }

      // returning explicit true
      {
        const numbers: number[] = [];
        await listEndpoint().autoPagingEach(async item => {
          numbers.push(item);
          return numbers.length < 3;
        });
        expect(numbers).toMatchSnapshot();
      }
    });

    test('returns result from first request when not using auto-paging', async done => {
      const listEndpoint = createMockedListEndpoint();
      const response = await listEndpoint();
      expect(response).toMatchSnapshot();
      expect(response.items).toBeInstanceOf(Array);

      listEndpoint().then(res => {
        expect(res).toMatchSnapshot();
        done();
      });
    });
  });
});

async function chunkingTester(
  axiosMock: MockAdapter,
  metadataMap: MetadataMap,
  url: string,
  endpoint: (items: unknown[]) => Promise<any>
) {
  const items = fillArray();
  const chunkCounter = countChunks(axiosMock, url);
  const response = await endpoint(items);
  expect(chunkCounter.value).toBe(3);
  expect(metadataMap.get(response)).toBeDefined();
  return { items, response };
}
