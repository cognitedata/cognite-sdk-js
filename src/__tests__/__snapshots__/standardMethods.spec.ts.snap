// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`standard methods generateListEndpoint authPagingForEach 1`] = `
Array [
  0,
  1,
  2,
]
`;

exports[`standard methods generateListEndpoint authPagingForEach 2`] = `
Array [
  0,
  1,
  2,
]
`;

exports[`standard methods generateListEndpoint authPagingForEach 3`] = `
Array [
  0,
  1,
  2,
]
`;

exports[`standard methods generateListEndpoint autoPagingToArray 1`] = `
Array [
  0,
]
`;

exports[`standard methods generateListEndpoint autoPagingToArray 2`] = `
Array [
  0,
  1,
  2,
]
`;

exports[`standard methods generateListEndpoint autoPagingToArray 3`] = `
Array [
  0,
  1,
  2,
  3,
  4,
  5,
  6,
  7,
  8,
  9,
  10,
  11,
  12,
]
`;

exports[`standard methods generateListEndpoint for-await 1`] = `
Array [
  0,
  1,
  2,
  3,
  4,
  5,
  6,
  7,
  8,
  9,
  10,
  11,
  12,
]
`;

exports[`standard methods generateListEndpoint returns result from first request when not using auto-paging 1`] = `
Object {
  "items": Array [
    0,
    1,
  ],
  "next": [Function],
  "nextCursor": "2",
}
`;

exports[`standard methods generateListEndpoint returns result from first request when not using auto-paging 2`] = `
Object {
  "items": Array [
    0,
    1,
  ],
  "next": [Function],
  "nextCursor": "2",
}
`;
